www/front_src/src/redux/selectors/navigation.js

```diff
@@ -0,0 +1,129 @@
+import { createSelector } from 'reselect';
+
+// loop on each child and get first available url
+function getFirstAvailableUrl(item) {
```

Comment:
I feel like we could make the flow of this function easier to follow if we divide into sub-functions.

What about:
```suggestion
function findFirstUrl(item) {
  if (item.groups) {
    const groupWithUrl = item.groups.find(findFirstUrl);

    return groupWithUrl ? findFirstUrlInChildren(groupWithUrl) : undefined;
  }

  return item.children ?  findFirstUrlInChildren(item) : undefined;

}

function findFirstUrlInChildren(item) {
  const childWithUrl = item.children.find((child) => child.url);

  return childWithUrl ? getUrl(childWithUrl) : undefined;
}

function getUrl(item) {
  return item.is_react ? item.url : `/main.php?p=${item.page}${item.options}`; 
}
```
It would be great to add some unit tests for that as well